src/Libraries/OrchardCore.Commerce.MoneyDataType/Serialization/LegacyAmountConverter.cs

```diff
@@ -78,4 +83,15 @@ public override void WriteJson(JsonWriter writer, Amount amount, JsonSerializer
 
         writer.WriteEndObject();
     }
+
+    public static Amount ReadString(string text)
+    {
+        var parts = text.Split();
+        if (parts.Length < 2) throw new InvalidOperationException($"Unable to parse string amount \"{text}\".");
```

Comment:
Yeah, it was intentional. By doing `string.Join` with an empty separator we remove all spaces after the first anyway. So you can enter "HUF 1 000 000" and it will still parse. It's free ~real estate~ readability!